include/ews/ews.hpp

```diff
@@ -8092,6 +8153,62 @@ static_assert(std::is_move_constructible<mailbox>::value, "");
 static_assert(std::is_move_assignable<mailbox>::value, "");
 #endif
 
+struct directory_id
+{
+    directory_id() {}
+    directory_id(std::string idstring) { id = idstring; }
+    std::string id;
+};
+
+#ifdef EWS_HAS_NON_BUGGY_TYPE_TRAITS
+static_assert(std::is_default_constructible<directory_id>::value, "");
+static_assert(std::is_copy_constructible<directory_id>::value, "");
+static_assert(std::is_copy_assignable<directory_id>::value, "");
+static_assert(std::is_move_constructible<directory_id>::value, "");
+static_assert(std::is_move_assignable<directory_id>::value, "");
+#endif
+
+struct resolution
+{
+    ews::mailbox mailbox;
+    ews::directory_id directory_id;
+};
+
+#ifdef EWS_HAS_NON_BUGGY_TYPE_TRAITS
+static_assert(std::is_default_constructible<resolution>::value, "");
+static_assert(std::is_copy_constructible<resolution>::value, "");
+static_assert(std::is_copy_assignable<resolution>::value, "");
+static_assert(std::is_move_constructible<resolution>::value, "");
+static_assert(std::is_move_assignable<resolution>::value, "");
+#endif
+
+struct resolution_set
+{
+    resolution_set()
+    {
+        indexed_paging_offset = 0;
+        numerator_offset = 0;
+        absolute_denominator = 0;
+        includes_last_item_in_range = true;
+        total_items_in_view = 0;
+    }
+
+    int indexed_paging_offset;
+    int numerator_offset;
+    int absolute_denominator;
+    bool includes_last_item_in_range;
+    int total_items_in_view;
+    std::vector<resolution> resolutions;
```

Comment:
Maybe you want to consider the size of the type and the order of it's individual members. Take a look at 

1. http://www.catb.org/esr/structure-packing/ 
2. https://eli.thegreenplace.net/2012/12/17/dumping-a-c-objects-memory-layout-with-clang